omod/src/main/java/org/openmrs/module/appointments/web/mapper/SingleAppointmentRecurringPatternMapper.java

```diff
@@ -0,0 +1,90 @@
+package org.openmrs.module.appointments.web.mapper;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.openmrs.api.APIException;
+import org.openmrs.module.appointments.helper.AppointmentServiceHelper;
+import org.openmrs.module.appointments.model.Appointment;
+import org.openmrs.module.appointments.model.AppointmentAudit;
+import org.openmrs.module.appointments.model.AppointmentRecurringPattern;
+import org.openmrs.module.appointments.service.AppointmentsService;
+import org.openmrs.module.appointments.web.contract.AppointmentRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+
+@Component
+@Qualifier("singleAppointmentRecurringPatternMapper")
+public class SingleAppointmentRecurringPatternMapper extends AbstractAppointmentRecurringPatternMapper {
+
+    @Autowired
+    private AppointmentsService appointmentsService;
+
+    @Autowired
+    private AppointmentMapper appointmentMapper;
+
+    @Autowired
+    private AppointmentServiceHelper appointmentServiceHelper;
+
+    @Override
+    public AppointmentRecurringPattern fromRequest(AppointmentRequest appointmentRequest){
+        String uuid = appointmentRequest.getUuid();
+        Appointment appointment = appointmentsService.getAppointmentByUuid(uuid);
+        Appointment newAppointment;
+        if(appointment==null)
```

Comment:
Can we use AppointmentValidator class here to validate the appointment ??

Reply:
Removed this part as we are doing this as part of editAppointmentRequestValidation.